diesel/src/connection/mod.rs

```diff
@@ -4,12 +4,18 @@ use query_source::Queryable;
 use result::*;
 use types::HasSqlType;
 
+/// To perform simple operations to the backed that implements this trait.
+/// `SimpleConnection` will be used for migrations. Every backend shall implement
+/// their own way to run migrations scripts.
 pub trait SimpleConnection {
-    #[doc(hidden)]
+    /// Migrations will use this function to run the `up.sql` and `down.sql` files.
+    /// New backends shall provide their own mechanism to execute the migration SQL scripts
     fn batch_execute(&self, query: &str) -> QueryResult<()>;
 }
 
+/// Every backend shall implement this trait to perform connections to the database.
```

Comment:
Can this be phrased as a description of what the trait does, rather than a command?